selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/CreateFactoryFromUiWithKeepDirTest.java

```diff
@@ -79,13 +79,16 @@
 
   @BeforeClass
   public void setUp() throws Exception {
-    ide.open(workspace);
+    ide.open(testWorkspace);
   }
 
   @AfterClass
   public void tearDown() throws Exception {
     if (factoryWsName != null) {
```

Comment:
To figure out the workspace you should request factory by name and get workspace name from config.

Reply:
I see. I will rewrite the tests to get new workspace name from factory configuration through REST API.

Reply:
Well, it seems it's impossible to get workspace name from factory configuration so that it contains name of workspace which factory created from. But, actual name of workspace created form factory can't be the same as workspace of "parent" workspace and Che creates new workspace with suffix "_1":
```javascript
{  
   "v":"4.0",
   "name":"keepFactorybk",
   "workspace":{  
      "environments":{  
         "workspacejeqfa3":{  
...
      },
      "defaultEnv":"workspacejeqfa3",
      "projects":[  
         {  
            "links":[  

            ],
            "name":"FactoryFromUiWithKeepDirTestne",
            "attributes":{  
               "contribute_to_branch":[  
                  "master"
               ]
            },
...
         },
...

      ],
      "links":[  

      ]
   },
   "id":"factory0wl3dz5z8e1vgc8z",
   "creator":{  
      "name":"che",
      "created":1505924420430,
      "email":"che@eclipse.org",
      "userId":"che"
   },
   "links":[  
    ]
}
```

Reply:
Another way is to set name of new workspace directly in factory configuration, but it makes tests a bit harder to understand.

Reply:
Yet another way is to create dedicated user for each test class and in this case we could remove all workspaces which test creates without knowledge of the names of workspaces. But it will require much more time so I propose to stay this changes as it meanwhile.

Reply:
TestFactory#delete encapsulate deletion of the factory and workspaces.

https://github.com/eclipse/che/blob/3c9df765384a4703f7df0cba2003ed2c25d656dd/selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java#L70

To create TestFactory you need a factory url and invoke `TestFactoryInitializer#fromUrl` method.

Reply:
Good advice, thank you!